microsoft/testsuites/core/cpu.py

```diff
@@ -177,3 +257,95 @@ def cpu_count_check(self, node: Node, log: Logger) -> None:
             "The VM may end up being incorrectly configured on some Azure hosts,"
             " it is a known host bug, please check the host version."
         ).is_equal_to(caculated_cpu_count)
+
+    @testsuite.TestCaseMetadata(
+        description="""
+            This test will verify if the CPUs inside a Linux VM are processing VMBus
+            interrupts by checking the /proc/interrupts file.
+
+            There are 3 types of Hyper-v interrupts : Hypervisor callback
+            interrupts, Hyper-V reenlightenment interrupts, and Hyper-V stimer0
+            interrupts.
+
+            Hyper-V reenlightenment interrupts are 0 unless the VM is doing migration.
+
+            Hypervisor callback interrupts are vmbus events that are generated on all
+            the vmbus channels, which belong to different vmbus devices. A VM with upto
+            4 vcpu on Azure/Hyper-V should have a NetVSC NIC, which normally has 4 VMBus
+            channel and should be bound to all the vCPUs.
+
+            Hyper-V Synthetic timer interrupts should be received on each CPU if the VM
+            is run for a long time. We can simulate this process by running CPU
+            intensive workload on each vCPU.
+
+            Steps:
+            1. Look for the Hyper-v timer property of each vCPU under /proc/interrupts
+            2. For Hyper-V reenlightenment interrupt, verify that the interrupt count
+            for all vCPU are zero.
+            3. For Hypervisor callback interrupt, verify that atleast min(#vCPU, 4)
+            vCPU's are processing interrupts.
+            4. For Hyper-V Synthetic timer, run a CPU intensive command on each vCPU and
+            verify that every vCPU is processing the interrupt.
+            """,
+        priority=2,
+    )
+    def verify_vmbus_interrupts(self, node: Node, log: Logger) -> None:
+        found_hyperv_interrupt = False
+        cpu_count = node.tools[Lscpu].get_core_count()
+        log.debug(f"{cpu_count} CPU cores detected...")
+
+        self._create_stimer_interrupts(node, cpu_count)
+        interrupts = self._get_interrupt_data(node)
+        for interrupt in interrupts:
+            is_hyperv_interrupt = any(
+                [(substr in interrupt.metadata) for substr in hyperv_interrupt_substr]
+            )
+            if not is_hyperv_interrupt:
+                continue
+            log.debug(f"Processing Hyper-V interrupt : {interrupt}")
+            assert_that(
+                len(interrupt.interrupt_count),
+                "Interrupt count should be present for each cpu.",
+            ).is_equal_to(cpu_count)
+            if interrupt.irq_number == "HRE" or "reenlightenment" in interrupt.metadata:
+                assert_that(
+                    all(
```

Comment:
It looks the assertion logic is the same, only message is different. So the logic can be moved after the logic. The logic only set messages.